tree/tree/src/TTree.cxx

```diff
@@ -8212,19 +8212,29 @@ void TTree::ResetAfterMerge(TFileMergeInfo *info)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Tell a branch to set its address to zero.
+/// Tell a branch to set its address to zero. This is also done for all
+/// trees in the list of clones.
 ///
 /// @note If the branch owns any objects, they are deleted.
 
 void TTree::ResetBranchAddress(TBranch *br)
 {
    if (br && br->GetTree()) {
       br->ResetAddress();
+      if (GetListOfClones()) {
+         for (TObjLink *lnk = GetListOfClones()->FirstLink(); lnk; lnk = lnk->Next()) {
+            TTree *clone = (TTree *)lnk->GetObject();
+            auto clbr = clone ? clone->FindBranch(br->GetName()) : nullptr;
+            if (clbr)
+               clone->ResetBranchAddress(clbr);
```

Comment:
We need to call here the subset of `CopyAddresses` related to the branch and it leafs.  This will require a bit of refactoring of `CopyAddresses` :(